src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.ts

```diff
@@ -906,8 +917,36 @@ export class PoolFormComponent extends CdForm implements OnInit {
           }
           this.form.setErrors({ cdSubmitButton: true });
         },
-        complete: () => this.router.navigate(['/pool'])
+        complete: () => this.runMirrorModeTask(poolName, isPoolTypeRbd)
+      });
+  }
+
+  private runMirrorModeTask(poolName: string, isPoolTypeRbd: boolean) {
+    const mirroringEnabled = this.form.getValue('mirroringEnabled');
+    if (isPoolTypeRbd) {
+      const request = new PoolEditModeResponseModel();
+      if (mirroringEnabled) {
+        request.mirror_mode = 'pool';
+      } else {
+        request.mirror_mode = 'disabled';
+      }
+      const action = this.taskWrapper.wrapTaskAroundCall({
+        task: new FinishedTask('rbd/mirroring/pool/edit', {
+          pool_name: poolName
+        }),
+        call: this.rbdMirroringService.updatePool(poolName, request)
+      });
+
+      action.subscribe({
+        error: () => this.form.setErrors({ cdSubmitButton: true }),
+        complete: () => {
+          this.rbdMirroringService.refresh();
+          this.router.navigate(['/pool']);
+        }
```

Comment:
you are invoking a new taskWrapper inside another taskWrapper and I don't see the point of it. This task is not relavant to the user since what user expects to be done is the pool creation to be successful and not about the background tasks..

In anycase if we do it at the API level, we don't need it.

Reply:
My thoughts: I think we can have different APIs it would be better as it replicates how CLI behaves as well and admin user can understand it bu looking at cli parameters.
We can leave this task of automation to a extended layer which is UI in our case and CLI wrapper(in-future).

But yes, task manager doesn't needs to be invoked, as user doesn't need to be informed separately about it.